server/src/cores/access.core.ts

```diff
@@ -142,6 +144,12 @@ export class AccessCore {
           : new Set();
       }
 
+      case Permission.ALBUM_WRITE: {
+        return sharedLink.allowUpload
+          ? await this.repository.album.checkSharedLinkAccess(sharedLinkId, ids)
+          : new Set();
+      }
+
```

Comment:
I don't think the album details should should be updated via share link authentication.

Reply:
I believe this doesn't allow that. This is `ALBUM_WRITE`, which gives the shared link, if `allowUpload` has been enabled, the ability to add photos to the album. Modifying album details are behind `ALBUM_UPDATE` as seen here:

https://github.com/mgabor3141/immich/blob/cecb99b4b6edadcb3e4e841df4a83318f580db04/server/src/services/album.service.ts#L139